src/diff.jl

```diff
@@ -0,0 +1,137 @@
+using ZXCalculus: contains, dagger, concat!
+
+"""
+Take derivative of ZXWDiagram with respect to a parameter
+
+Assuming Spiders have Parameter of type PiUnit which is parameterized purely by θ
+"""
+function diff_diagram(zxwd::ZXWDiagram{T,P}, θ::Symbol) where {T,P}
+    add_global_phase!(zxwd, P(π / 2))
```

Comment:
Do we need to match the pattern and check if there are spiders containing $\theta$ first before manipulating the diagram?

Reply:
Sorry, I overlooked this point. But a simple checking length of `vs` on next line should suffice. That function is responsible for finding all $\theta$ and $-\theta$.